### DISCUSSION TITLE
Questions about the roslyn version

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/roslyn#50395] [Q&A] [answered]
In this page wrote :
[https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version)
"C# 8.0 is supported only on .NET Core 3.x and newer versions" ...

But in nuget Microsoft.Net.Compilers page wrote :
[https://www.nuget.org/packages/Microsoft.Net.Compilers/](https://www.nuget.org/packages/Microsoft.Net.Compilers/)
"Dependencies : .NETFramework 4.7.2"

Well, can I use C# 8 (Roslyn 3.9) in .NET Framework 4.8 at Windows Server 2016?

**Comment by PathogenDavid** [comment, marked as answer]
The actual compiler will run on .NET Framework 4.7.2, but newer language features are only officially supported with newer runtimes. (That being said, many of them work despite the lack of official support. Anything that depends on runtime features does work.)

**Reply by ghost** [reply]
Thank you for your reply, but I think Microsoft's official documents should be written more clearly.

**Reply by PathogenDavid** [reply]
No problem! It definitely seems to be a common point of confusion. You can leave feedback on the documentation using the button at the bottom of the page.
